sphinx/config.py

```diff
@@ -350,19 +298,50 @@ def __contains__(self, name):
         return name in self.values
 
     def __iter__(self):
-        # type: () -> Iterable[ConfigValue]
+        # type: () -> Generator[ConfigValue, None, None]
         for name, value in iteritems(self.values):
             yield ConfigValue(name, getattr(self, name), value[1])  # type: ignore
 
     def add(self, name, default, rebuild, types):
         # type: (unicode, Any, Union[bool, unicode], Any) -> None
-        self.values[name] = (default, rebuild, types)
+        if name in self.values:
+            raise ExtensionError(__('Config value %r already present') % name)
+        else:
+            self.values[name] = (default, rebuild, types)
 
     def filter(self, rebuild):
         # type: (Union[unicode, List[unicode]]) -> Iterator[ConfigValue]
         if isinstance(rebuild, string_types):
             rebuild = [rebuild]
-        return (value for value in self if value.rebuild in rebuild)  # type: ignore
+        return (value for value in self if value.rebuild in rebuild)
+
+
+def eval_config_file(confdir, filename, tags):
```

Comment:
Same comment here. You could just replace `confdir` and `filename` with `path` or similar. This seems like an improvement to me, anyway.